### PR TITLE
Reduce the number of intra modes searched for chroma

### PR DESCRIPTION
**Pull request by fbossen**
Limit the intra modes searched for chroma to the colocated luma
mode plus the DC mode if different from the first mode

### CODE REVIEW

File: src/rdo.rs [hunk 1]
```diff
@@ -214,15 +214,14 @@ pub fn rdo_mode_decision(
...
-    };
+    let mut mode_set_chroma = vec![ luma_mode ];
+
+    if is_chroma_block && luma_mode.is_intra() && luma_mode != PredictionMode::DC_PRED {
+      mode_set_chroma.push(PredictionMode::DC_PRED);
+    }
 
     // Find the best chroma prediction mode for the current luma prediction mode
-    for &chroma_mode in mode_set_chroma {
+    for &chroma_mode in &mode_set_chroma {
```
Comment (ycho) [inline src/rdo.rs]: For given luma_mode, it now checks only one chroma_mode, so it does not need for loop here?

Comment (fbossen) [reply src/rdo.rs]: Sometimes there should be a second mode (see push() on line 220)

**Comment by ycho** [comment]
Related to the goal of this patch (i.e. improving speed), I again figure out that if more than one chroma intra modes are tested as previously, luma prediction in encode_block_b() is called duplicated among different chroma intra modes though luma work as context for chroma?

File: src/rdo.rs (hunk 1 above)
Comment (ycho) [reply src/rdo.rs]: Oh right- you mentioned it in the commit message as well. If luma is not DC then chroma rdo loop test DC and the non-DC mode.

**Comment by fbossen** [comment]
@ycho Yes, I recently opened #463 on this topic

**Comment by fbossen** [comment]
At default speed setting (3):

|    PSNR | PSNR Cb | PSNR Cr | PSNR HVS |    SSIM | MS SSIM | CIEDE 2000 |
|    ---: |    ---: |    ---: |     ---: |    ---: |    ---: |       ---: |
| -0.5450 |  1.1756 |  1.2647 |  -0.7288 | -0.4745 | -0.5919 |    -0.0551 |

**Comment by fbossen** [comment]
Runs about 40% at default speed setting

**Comment by ycho** [comment]
That's nice speed up. Also, I am sure that improving Y psnr while losing small in UV will be much appreciated by @tterribe.

**Pull request merged**